src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -22,16 +24,24 @@ class LeonardoService(gdDAO: DataprocDAO, dbRef: DbReference)(implicit val execu
   }
 
   def createCluster(googleProject: String, clusterName: String, clusterRequest: ClusterRequest): Future[Cluster] = {
-    gdDAO.createCluster(googleProject, clusterName, clusterRequest) flatMap { clusterResponse: ClusterResponse =>
+    def create = gdDAO.createCluster(googleProject, clusterName, clusterRequest) flatMap { clusterResponse: ClusterResponse =>
       dbRef.inTransaction { dataAccess =>
         dataAccess.clusterQuery.save(Cluster(clusterRequest, clusterResponse))
       }
     }
+
+    dbRef.inTransaction { dataAccess =>
```

Comment:
with the code structured like this, your call to `gdDAO.createCluster` and your db transaction here will run in parallel. i suspect you want to chain them together, something like `create.flatMap { _ => db.inTxn }`